workers/data_refinery_workers/processors/illumina.py

```diff
@@ -0,0 +1,259 @@
+            job_context["success"] = False
+            return job_context
+        else:
+            job_context['probeId'] = headers[0]
+
+        # Then the detection Pvalue string, which is always(?) some form of 'Detection Pval'
+        for header in headers:
+            if 'DETECTION_PVAL' in header.upper().replace(' ', '_'):
+                # Could be <SAMPLE_ID>.Detection Pval, etc
+                if '.' in header:
+                    job_context['detectionPval'] = header.split('.')[-1]
+                else:
+                    job_context['detectionPval'] = header
+                break
+        else:
+            job_context["job"].failure_reason = "Could not detect PValue column!"
+            job_context["success"] = False
+            return job_context
+
+        # Then, finally, create an absolutely bonkers regular expression
+        # which will explictly hit on any sample which contains a sample
+        # ID _and_ ignores the magical word 'BEAD', etc. Great!
+        column_ids = ""
+        for sample in job_context['samples']:
+            for offset, header in enumerate(headers, start=1):
+
+                if sample.title == header:
+                    column_ids = column_ids + str(offset) + "," 
+                    continue
+                if header.upper().replace(' ', '_') == "RAW_VALUE":
+                    column_ids = column_ids + str(offset) + "," 
+                    continue
+                if sample.title in header and \
+                'BEAD' not in header.upper() and \
+                'NARRAYS' not in header.upper() and \
+                'ARRAY_STDEV' not in header.upper() and \
+                'PVAL' not in header.upper().replace(' ', '').replace('_', ''):
+                    column_ids = column_ids + str(offset) + "," 
+                    continue
+        for offset, header in enumerate(headers, start=1):
+            if 'AVG_Signal' in header:
+                column_ids = column_ids + str(offset) + "," 
+                continue
+
+        column_ids = column_ids[:-1]
+        job_context['columnIds'] = column_ids
+    except Exception as e:
+        job_context["job"].failure_reason = str(e)
+        job_context["success"] = False
+        return job_context
+
+    return job_context
+
+def _run_illumina(job_context: Dict) -> Dict:
+    """Processes an input TXT file to an output PCL file using a custom R script.
+    Expects a job_context which has been pre-populated with inputs, outputs
+    and the column identifiers which the R script needs for processing.
+    """
+    input_file_path = job_context["input_file_path"]
+
+    # We need to detect the sub-platform. Looking for the chip version
+    # in the metadata is the best we can do for right now.
+    annotation = job_context['samples'][0].sampleannotation_set.all()[0]
+    annotation_data = str(annotation.data).encode('utf-8').upper()
+    if "V2".encode() in annotation_data or "V 2".encode() in annotation_data:
```

Comment:
Related to my concerns over on `workers/data_refinery_workers/processors/illumina.R` (line 396). If #222 was completed, what would this look like? (Can help inform how we do #222!) We could also consider the support for additional platforms out of the scope of the PR and file a ticket -- up to you.